src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -299,17 +762,103 @@ IMF_STD_ATTRIBUTE_DEF (framesPerSecond, FramesPerSecond, Rational)
+// the encoding might be camera-vendor-specific, or it might be formally
+// or informally standardized (e.g. an ASC Frameline Decision List)
+//
+
+IMF_STD_ATTRIBUTE_DEF (framelines, Framelines, std::string)
+
+
+//
+// imageRotation -- rotation of the image in degrees, in the closed interval
+// [-45, +45], with the rotation around the center of the display window, a
+// positive value specifying a rotation from x-axis to y-axis
+//
+
+IMF_STD_ATTRIBUTE_DEF (imageRotation, ImageRotation, float)
+
+
+//
+// cameraPosition -- x, y, z position of the camera, in meters
+//
+// The attribute may record the center of the camera's sensor, or the
+// center of the entrance pupil of the lens. The coordinate system for
+// the camera's position is fixed in relation to the set, and is a
+// right-handed Cartesian coordinate system with the z-axis pointing
+// upwards and the y-axis pointing 90 degrees to the left of the
+// x-axis direction.
+//
+// Recording the position of the center of the entrance pupil provides the
+// most usable value in some circumstances, e.g. as the camera pans. This
+// position changes, however, with lens and focus. For cases when this
+// precision is not needed, recording the center of the sensor is provided
+// as an alternative.
+//
+// For higher precision the chosen coordinate system is usually NOT the GPS
+// system, but an on-set coordinate system. Any coordinate system can be 
+// chosen for recording the camera's position, as long as its axes are
+// orthogonal, and z points up. The coordinate system can be moving with
+// the set, for example, when on a ship. Its reference point (usually
+// assignged position 0,0,0) can be within or outside the set.
+
+IMF_STD_ATTRIBUTE_DEF (cameraPosition, CameraPosition, IMATH_NAMESPACE::V3f)
+
+
+//
+// cameraUpDirection -- the 'up' direction of the camera
+//
+// If present, must be accompanied by a cameraViewingDirection attribute.
+// The 'up' direction should be perpendicular to the direction specified in the
+// cameraViewingDirection attribute. The coordinate system in which the
+// 'up' direction is expressed is that defined for the cameraViewingDirection
+// attribute.
+
+IMF_STD_ATTRIBUTE_DEF (cameraUpDirection, CameraUpDirection, IMATH_NAMESPACE::V3f)
+
+
+//
+// cameraViewingDirection -- the viewing direction of the camera, i.e. the
+// direction of the optical axis of the lens, as a right-handed Cartesian
+// coordinate system with the z-axis pointing upwards and the y-axis
+// pointing 90 degrees to the left of the x-axis direction.
+// 
+// If present, must be accompanied by a cameraUpDirection attribute.
+//
+// If the cameraPosition attribute is present, the axes of the
+// cameraViewingDirection attribute are parallel to, and point in the same
+// direction as, those of the cameraPosition attribute; if the cameraPosition
+// attribute is not present then the directions of the axes in the coordinate
+// shall be fixed in relation to the set.
```

Comment:
Does this actually make sense? whether or not the cameraPosition attribute is indicated, I don't see how that can actually modify the direction.

Reply:
It helps to look at the two definitions next to each other:

![image](https://user-images.githubusercontent.com/1134845/103468108-b45b2b80-4d23-11eb-985a-ac77fe3d17dc.png)

![image](https://user-images.githubusercontent.com/1134845/103468111-c046ed80-4d23-11eb-9ed7-90ed9c6c969f.png)

For both camera position and viewing direction there is a desire to relate those to the environment, which is what the "fixed in relation to the set" thing is about. If viewingDirection didn't mention cameraPosition, one could conceivably misread cameraPosition and viewingDirection as being allowed to define independent coordinate systems (well, not totally independent as Z has to point 'upwards' in some global sense). I think that the final paragraph there is trying to prohibit that interpretation but is perhaps phrased in a way that is confusing more than helpful.

How about

_The cameraViewingDirection coordinate system shares the same fixed relationship to the set as does the cameraPosition coordinate system, if the latter is present. In the absence of a cameraPosition coordinate system, the cameraViewingDirection coordinate system establishes itself a fixed relationship to the set._

I will check with the 2065-4 revision group to see if all the above is accurate, and if so, will not only change the code but will change the 2065-4:202x draft.

Assuming they agree with my analysis and language, would replacing the last paragraph of cameraViewingDirection's comment block resolve your comment?